### PR TITLE
python3Packages.monty: add missing pydanctic and tqdm dependencies

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://nix-cache.s3.amazonaws.com/log/jlz2mql8x2v3smdy2ccyh5g47cn378p2-python3.8-monty-2021.3.3.drv

ZHF: #122042 

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Looked into `python3Packages.sumo` but we are missing `castepxbin==0.1.0`.

> ERROR: Could not find a version that satisfies the requirement castepxbin==0.1.0 (from sumo)
> ERROR: No matching distribution found for castepxbin==0.1.0

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 122097` at 3f508d85 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>python38Packages.dftfit</li>
    <li>python38Packages.lammps-cython</li>
    <li>python38Packages.pymatgen-lammps</li>
    <li>python39Packages.lammps-cython</li>
    <li>python39Packages.pymatgen-lammps</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/fb4f06c66cf36a2c3fc86dca1e404e9c">python38Packages.sumo</a></li>
    <li><a href="https://gist.github.com/7191422b9f2897251fe09485d9adedcf">python39Packages.sumo</a></li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.monty</li>
    <li>python38Packages.pymatgen</li>
    <li>python39Packages.monty</li>
    <li>python39Packages.pymatgen</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/python-modules/monty/default.nix [hunk 1]
```diff
@@ -27,14 +29,7 @@ buildPythonPackage rec {
   };
 
   checkInputs = [ lsof nose numpy msgpack coverage coveralls pymongo];
```
[the remark this answers is not in the document]
Comment (mweinelt) [reply pkgs/development/python-modules/monty/default.nix:31]: https://github.com/materialsvirtuallab/monty/search?q=tqdm

tqdm is used outside of tests.

[the remark this answers is not in the document]
Comment (dotlambda) [reply pkgs/development/python-modules/monty/default.nix:31]: True. I thought I had searched for both.

**Comment by mweinelt** [comment]
Did another pass to weed out unused dependencies. Their requirements.txt is a mess.

**Review by jonringer — APPROVED** [review]
LGTM

```
https://github.com/NixOS/nixpkgs/pull/122097

5 packages marked as broken and skipped:
python38Packages.dftfit python38Packages.lammps-cython python38Packages.pymatgen-lammps python39Packages.lammps-cython python39Packages.pymatgen-lammps

2 packages failed to build:
python38Packages.sumo python39Packages.sumo

4 packages built:
python38Packages.monty python38Packages.pymatgen python39Packages.monty python39Packages.pymatgen
```

**Comment by dotlambda** [comment]
https://github.com/NixOS/nixpkgs/pull/122161

**Comment by dotlambda** [comment]
> Looked into `python3Packages.sumo` but we are missing `castepxbin==0.1.0`.

https://github.com/NixOS/nixpkgs/pull/122162

**Pull request merged**